pulp_rpm/app/modulemd.py

```diff
@@ -50,130 +58,121 @@ def modules_packages(modules):
     version.add_content(Package.objects.filter(pk__in=packages_to_add))
 
 
-def parse_modulemd(module_names, module_index):
+def split_modulmd_file(file):
     """
-    Get modulemd NSVCA, artifacts, dependencies.
-
-    Args:
-        module_names (list):
-            list of modulemd names
-        module_index (mmdlib.ModuleIndex):
-            libmodulemd index object
+    Helper method to preserve original formatting of modulemd.
+    """
+    m_open = gzip.open if file.url.endswith(".gz") else open
+    with m_open(file.path, "r") as modulemd_file:
+        module = ""
+        for line in modulemd_file:
+            line = line.decode("utf-8")
+            module += line
+            if line.startswith("..."):
```

Comment:
Are they 100% guaranteed by yaml specification to be separated by `...`?

Reply:
It is specified, but optional, better to split by ``---`` as document prefix

Reply:
I'm a bit worried about having any expectations that an optional separator will exist

Reply:
now its switched for the mandatory one